### PR TITLE
ci: add cli k8s compatibility table artifact upload to ci

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- Add uploading of CLI / K8s version compatibility information to the CI pipeline to satisfy the API specified in [this RFC](https://github.com/edgelesssys/constellation/blob/main/rfc/cli-api.md)
- [Successful run](https://github.com/edgelesssys/constellation/actions)

### Additional info
- A refactoring might be done at some point to not have the cli info upload in the image build pipeline for a better structure, but I considered it as not being worth it at the moment

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

**Review by katexochen — COMMENTED** [review]
To conclude our face to face discussion:

- We can implement the cli info in the versionsapi package, including the functions that the related structs have
- We can add a fetch method to the versionsapi/fetcher and fetch and update method to the versionsapi/client
- The versionsapi fetcher can be used in the constellation CLI to get the info struct from our API
- The versionsapi client can be used in the hack tool to upload the info struct to the API. We can pass needed information per flag so there isn't a need to manipulate using jq.
- We likely should also do the same with the image info struct in the future.

File: internal/versionsapi/cliinfo.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	url.Path = c.JSONPath()
+	return url.String(), nil
+}
+
+// ValidateRequest validates the request parameters of the list.
+// The Kubernetes slice must be empty.
+func (c CLIInfo) ValidateRequest() error {
+	var retErr error
+	if err := ValidateRef(c.Ref); err != nil {
+		retErr = multierr.Append(retErr, err)
```
Comment (katexochen) [inline internal/versionsapi/cliinfo.go:59]:
```suggestion
        retErr = errors.Join(retErr, err)
```
we no longer use multierr, but the new native way of joining errors

File: .github/workflows/build-os-image.yml [hunk 2]
```diff
@@ -912,8 +914,32 @@ jobs:
...
+            > info.json
+
+      - name: Upload CLI compatibility information to S3
+        shell: bash
+        run: |
+          aws s3 cp \
+            "info.json" \
+            "s3://cdn-constellation-backend/${{ needs.build-settings.outputs.cliApiBasePath }}/info.json" \
+            --no-progress
+          echo -e "[CLI compatibility information](https://cdn.confidential.cloud/${{ needs.build-settings.outputs.cliApiBasePath }}/info.json)" >> "$GITHUB_STEP_SUMMARY"
```
Comment (katexochen) [inline .github/workflows/build-os-image.yml:938]: You can move this step into the Go hack tool, too. Use the versionsapi/client update function to upload.

Comment (msanft) [reply .github/workflows/build-os-image.yml]: see our teams conv regarding this

**Comment by msanft** [comment]
[Successful run](https://github.com/edgelesssys/constellation/actions/runs/4243449371/jobs/7376982181)

**Comment by msanft** [comment]
[Successful run after changes](https://github.com/edgelesssys/constellation/actions/runs/4250948093)

File: hack/cli-k8s-compatibility/main.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	if err != nil {
+		log.Fatalf("creating s3 client: %w", err)
+	}
+	defer func() {
+		if err := c.InvalidateCache(ctx); err != nil {
+			log.Fatalf("invalidating cache: %w", err)
+		}
+	}()
+
+	if err := c.UpdateCLIInfo(context.Background(), cliInfo); err != nil {
```
Comment (katexochen) [inline hack/cli-k8s-compatibility/main.go:63]:
```suggestion
	if err := c.UpdateCLIInfo(ctx, cliInfo); err != nil {
```

**Review by katexochen — APPROVED** [review]
lgtm

**Pull request merged**